pkg/webui/console/views/application-general-settings/index.js

```diff
@@ -80,24 +114,90 @@ const validationSchema = Yup.object().shape({
 })
 
 @connect(
-  state => ({
-    appId: selectSelectedApplicationId(state),
-    application: selectSelectedApplication(state),
-  }),
+  state => {
+    const mayViewApiKeys = checkFromState(mayViewOrEditApplicationApiKeys, state)
+    const mayViewCollaborators = checkFromState(mayViewOrEditApplicationCollaborators, state)
+    const apiKeysCount = selectApiKeysTotalCount(state)
+    const collaboratorsCount = selectCollaboratorsTotalCount(state)
+    const webhooksCount = selectWebhooksTotalCount(state)
+    const pubsubsCount = selectPubsubsTotalCount(state)
+    const mayPurgeApp = checkFromState(mayPurgeEntities, state)
+    const mayDeleteApp = checkFromState(mayDeleteApplication, state)
+
+    const entitiesFetching =
+      selectApiKeysFetching(state) ||
+      selectCollaboratorsFetching(state) ||
+      selectPubsubsFetching(state) ||
+      selectWebhooksFetching(state)
+    const error =
+      selectApiKeysError(state) ||
+      selectCollaboratorsError(state) ||
+      selectPubsubsError(state) ||
+      selectWebhooksError(state)
+
+    const fetching =
+      entitiesFetching ||
+      (mayViewApiKeys && typeof apiKeysCount === undefined) ||
+      (mayViewCollaborators && collaboratorsCount === undefined) ||
+      typeof collaboratorsCount === undefined ||
+      typeof pubsubsCount === undefined
+    const hasIntegrations = webhooksCount > 0 || pubsubsCount > 0
+    const hasApiKeys = apiKeysCount > 0
+    // Note: there is always at least one collaborator.
+    const hasAddedCollaborators = collaboratorsCount > 1
+    const isPristine = !hasApiKeys && !hasAddedCollaborators && !hasIntegrations
```

Comment:
What about devices?

Reply:
You cant delete/purge application with devices